### ISSUE TITLE
a module to create Keystone endpoints

### ISSUE THREAD

**Issue by jackivanov** [ansible/ansible#16809]
##### ISSUE TYPE
- Feature Idea
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Debian Jessie
##### SUMMARY

I found `os_user` module and a lot of other OS modules, but I can't find almost the most important module to work with Keystone. 
Is there a way to create Keystone endpoints?
##### EXPECTED RESULTS

I would like to use a native Ansible module to create Keystone endpoints. Something like this:

```
- name: Keystone | Create nova endpoints
  os_keystone_endpoints:
    state: present
    cloud: devstack
    name: nova
    type: compute
    public: http://10.90.180.208:8774/v2.1/%(tenant_id)s
    admin: http://10.90.180.208:8774/v2.1/%(tenant_id)s
    internal: http://10.90.180.208:8774/v2.1/%(tenant_id)s
```

Is that possible?
##### ACTUAL RESULTS

Now I'm using `shell` module and `openstack endpoint` commands, but it is not a good way

Thank you.

**Comment by alikins** [comment]
http://docs.ansible.com/ansible/list_of_cloud_modules.html#openstack has the list of open stack modules. For keystone, looks like there a few related modules:

http://docs.ansible.com/ansible/os_keystone_domain_module.html
http://docs.ansible.com/ansible/os_keystone_service_module.html
http://docs.ansible.com/ansible/os_keystone_role_module.html
http://docs.ansible.com/ansible/keystone_user_module.html
http://docs.ansible.com/ansible/os_keystone_domain_facts_module.html

This appears to be a user question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.
- IRC: #ansible on irc.freenode.net  
- mailing list: https://groups.google.com/forum/#!forum/ansible-project
  
  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

**Comment by kallaics** [comment]
Dear Alikins,

I have a same issue with Ansible 2.2.0. I read the documents more times, but I didn't found any possibility how can I create endpoint with these Openstack modules that you listed above.
I am using the keystone_user option, but it is deprecated, so I would like to change it.

Could you please write an quick example how can we define the endpoints for the services?

Many thanks!

Best regards,
   kallaics

**Comment by xiaosuiba** [comment]
Dear Alikins,

I have the same issue with the latest version. I'm now pretty sure their's no way to set endpoints of an openstack project. I only found a [third-party ansible](https://github.com/openstack-ansible/openstack-ansible-modules) module to do that thing like this:
```
keystone_service: token=$admin_token name=keystone type=identity description="Identity Service" public_url="http://192.168.206.130:5000/v2.0" internal_url="http://192.168.206.130:5000/v2.0" admin_url="http://192.168.206.130:35357/v2.0"
```

Could you please take a look at that project and add this feature to the next release version?

Thanks!
Best regards,
lijingjing

**Comment by ashak** [comment]
I'm trying to configure an OpenStack setup using ansible. I have got as far as creating the service endpoints for the identity service.. this would involve you manually running something like:

`openstack endpoint create --region RegionOne image public http://controller:9292`

The ansible os_keystone_service module allows you to define the service, but then ansible has no way to to allow you to create the service endpoints.

I came across this issue whilst googling for the exact same thing myself.

@alikins I think you have misdiagnosed this as a support request, whereas it's a genuine feature request for something that ansible currently doesn't support. Is it possible for it to be reopened and considered as an actual feature request? Everything in the initial request is still entirely valid.
